### ISSUE TITLE
symlink on windows unable to launch due to library linking 

### ISSUE THREAD

**Issue by musm** [neovim/neovim#17714]
### Neovim version (nvim -v)

NVIM v0.6.1

### Vim (not Nvim) behaves the same?

not sure

### Operating system/version

windows 11

### Terminal name/version

windows terminal

### $TERM environment variable

NA

### Installation

zip

### How to reproduce the issue

From a command prompt
```
mklink <desired location>\nvim.exe <folder>\Neovim\bin\nvim.exe
```

### Expected behavior

I'd expect the libraries to be found relative to the source directory. 

### Actual behavior

Basically all the libraries dll cannot be found when attempting to launch `nvim.exe`

This is frustrating for use cases where we want a central binary location, unfrountely we cannot symlink neovim due to the current bug report.

**Comment by yehorb** [comment]
Can confirm the same issue with the newest 0.7.0 release.
Neovim installed from zip. Link created with PowerShell: `New-Item -Type SymbolicLink -Target ..\nvim-win64\bin\nvim.exe -Path nvim.exe`. Executing the link produces no output.

**Comment by musm** [comment]
Oddly enough it works if you actually click  `nvim.exe` directly, just not from the terminal directly...

**Comment by justinmk** [comment]
This was reported with 0.6 which means probably not related to the cpack changes https://github.com/neovim/neovim/pull/17390

**Comment by 3N4N** [comment]
I don't know if it's known already but this problem only occurs for downloaded zipped releases.  If you build it from source, the symlink works fine, both for symlinks against 'build/bin/nvim.exe' or symlinks against the nvim binary in the extracted zip file created with `cpack -G ZIP -C Release`.

I'm building with clang of Msys whereas the github release is built with MSVC.  That might be a reason for this difference in behavior.  I don't have MSVC installed to check for that, unfortunately.

**Comment by erw7** [comment]
> I don't know if it's known already but this problem only occurs for downloaded zipped releases. If you build it from source, the symlink works fine, both for symlink against 'build/bin/nvim.exe' and symlink against the nvim binary in the extracted zip file created with cpack -G ZIP -C Release.

The reason this problem does not occur with the built `nvim.exe` symbolic link is probably due to the value of `PATH`. If you set the `PATH` value to `C:\WIndows;C:\Windows\System32`, the `nvim.exe` you built should have this problem as well as `zip`.

From the following document, it seems that the problem occurs because `7. the folder from which the application is loaded.` is a directory with a symbolic link. Therefore, a workaround is to create symbolic links for the necessary DLLs in the directory where the symbolic link was created(In addition, the `VIMRUNTIME` environment variable must be set).

Ref. https://learn.microsoft.com/en-us/windows/win32/dlls/dynamic-link-library-search-order

**Comment by erw7** [comment]
The problem to begin with seems to be that `luajit` are not statically linked.

To static link `luajit` we need to pass `static` as a command line argument in the following places. `INSTALL_COMMAND` should also be adjusted.

https://github.com/neovim/neovim/blob/46d4d420e56bb0b4aec696fd8164bffde02d2758/cmake.deps/cmake/BuildLuajit.cmake#L131

**Comment by dundargoc** [comment]
> The problem to begin with seems to be that `luajit` are not statically linked.
> 
> To static link `luajit` we need to pass `static` as a command line argument in the following places. And `INSTALL_COMMAND` should also be adjusted.
>  https://github.com/neovim/neovim/blob/46d4d420e56bb0b4aec696fd8164bffde02d2758/cmake.deps/cmake/BuildLuajit.cmake#L131

@erw7 mind sending a PR?

**Comment by 3N4N** [comment]
I recently learned about executable shims: https://docs.chocolatey.org/en-us/features/shim.

In one place, the article says, "[Shims] works better than symlinks. Symlinks on Windows fall down at file dependencies. So if your file depends on other files and DLLs, all of those need to also be linked."

I am not suggesting you stop fixing the problem described in this ticket.  I just thought this shimming concept may add some insight.
